### ISSUE TITLE
misc/wasm: incorrectly checks for require("fs") validity

### ISSUE THREAD

**Issue by tharvik** [golang/go#40446]
When loaded, `wasm_exec.js` tries to know on which platform it is run. "fs" is loaded if not already available and check if it is empty (when using webpack to provide an empty implementation) or not (when running inside node). The check is wrongly implemented as `Object.keys` returns an array of key, not a count, thus making that check always true, and unable to use with webpack.

The line in question
https://github.com/golang/go/blob/c4fed25553ee266ed9cb3a98e7a33a82af110ed4/misc/wasm/wasm_exec.js#L31

Which can be easily fixed by adding a `.length` after the `Object.keys(fs)`.

**Comment by agnivade** [comment]
That's right. There's a CL at https://go-review.googlesource.com/c/go/+/241119.

**Comment by tharvik** [comment]
Closing as duplicate then, thanks for the pointer.

**Comment by tharvik** [comment]
Oups, missread that,  reopening.

**Comment by neelance** [comment]
The CL will land in the next release cycle.
